presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java

```diff
@@ -274,7 +274,10 @@ private static void compareAndUpdateState(MethodHandle methodHandle, NullableDou
     {
         if (state.isNull()) {
             state.setNull(false);
-            state.setDouble(value);
+            // Not set value to NaN initially to ensure subsequent comparisons succeed
```

Comment:
I don't think this will work, as I think `SELECT min(a) from (VALUES NaN()) AS t(a);` should return `NaN`.

Reply:
Approach changed in latest commit to cover your described case :D